packages/host/app/components/card-catalog/modal.gts

```diff
@@ -213,44 +216,50 @@ export default class CardCatalogModal extends Component<Signature> {
     // returns all available realms and their cards that match a certain type criteria
     // realm filters and search key filter these groups of cards
     // filters dropdown menu will always display all available realms
-    if (this.currentRequest?.search.instancesByRealm.length) {
-      this.searchResults = this.currentRequest?.search.instancesByRealm;
+    if (this.request?.search.instancesByRealm.length) {
+      this.state.searchResults = this.request?.search.instancesByRealm;
     }
-    return this.currentRequest?.search.instancesByRealm ?? [];
+    return this.request?.search.instancesByRealm ?? [];
   }
 
   get displayedRealms(): RealmCards[] {
     // filters the available realm cards by selected realms
-    return this.selectedRealms.length
-      ? this.selectedRealms
+    return this.state.selectedRealms.length
+      ? this.state.selectedRealms
       : this.availableRealms;
   }
 
-  _selectedRealms = new TrackedArray<RealmCards>([]);
+  get request(): Request | undefined {
+    return this.state.request;
+  }
+
+  get state(): State {
+    if (this.stateStack.length <= 0) {
+      return {} as State;
+    }
 
-  get selectedRealms(): RealmCards[] {
-    return this._selectedRealms;
+    return this.stateStack[this.stateStack.length - 1];
   }
 
   @action onSelectRealm(realm: RealmCards) {
-    this._selectedRealms.push(realm);
+    this.state.selectedRealms.push(realm);
```

Comment:
why are we pushing a new item into a different state array--what happens if the length of the stateStack is not the same as the length of the selectedRealms array? i think the state stack is a good idea, but you should track all the state there--not in separate arrays that have to magically line up.

Reply:
I think this `selectedRealms` array is used by the users to define in which realms they want to search cards. So it's a part of the card catalog modal state.